### PR TITLE
Move Cross-Reference content to final homes

### PR DESCRIPTION
**Pull request by cwickham**
* Move out of Guide > Scholarly Writing in sidebar and collect into Guide > Cross References:
    Screenshot 2024-01-16 at 11 39 47 AM
    * [Basics](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references.html)
    * [Options](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-reference-options.html)
    * [Div Syntax](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references-divs.html)
    * [Custom Floats](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references-custom.html)
* Rewrite main cross-reference article, [Guide > Authoring > Cross-References > Basics](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references.html):
    * Reorder to cover reference syntax, and `\listof` first, then enumerate types
    * Identify figures, tables and code listings as float cross-references and link to div syntax and custom floats
    * Add basic div syntax
    * Move options to their own page
* Add links from [Guide > Authoring > Videos](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/videos.html#cross-references) and [Guide > Authoring > Diagrams](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/videos.html#cross-references) to [Guide > Cross-References > Div Syntax](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references-divs.html)
* Add `#rem` and `#sol` to [theorem types that can be cross-referenced](https://deploy-preview-974--mystifying-jepsen-fa4396.netlify.app/docs/authoring/cross-references.html#theorems-and-proofs)

### CODE REVIEW

**Comment by cwickham** [comment]
@jjallaire and @cscheid How do you feel about where I've put this in the sidebar nav? 

Current nav on left, proposed nav on right:
Screenshot 2024-01-16 at 12 00 59 PMScreenshot 2024-01-16 at 11 58 06 AM

Other options might be:

* Leave the main Cross References page where it was under **Guide > Scholarly Writing**, and put the other pages (Options, Div Syntax, and Custom Floats) under **Guide > Advanced > Cross References**
* Make **Guide > Scholarly Writing > Cross References** a section and nest the other pages under it.

File: docs/authoring/cross-reference-options.qmd [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+``` yaml
+---
+title: "My Document"
+crossref:
+  labels: alpha a        # (default is arabic)
+  subref-labels: roman i # (default is alpha a)
+---
+```
+
+If you would like, you can specify the number scheme for a specific type using the `*-labels` options. For example:
```
Comment (cscheid) [inline docs/authoring/cross-reference-options.qmd:69]: "numbering scheme"?

**Comment by cscheid** [comment]
I like the idea of flattening that hierarchy, but I do think it makes sense to make Cross-References a subsection of Scholarly Writing. I think I lean slightly towards the latter.

**Review by cscheid — APPROVED** [review]
Aside from one minor comment below, this is really great!

**Pull request merged**